apps/src/templates/AccessibleDialog.jsx

```diff
@@ -0,0 +1,25 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import FocusTrap from 'focus-trap-react';
+import CloseOnEscape from '@cdo/apps/templates/CloseOnEscape';
+import style from './accessible-dialogue.module.scss';
+
+const AccessibleDialog = ({onClose, children}) => (
+  <>
+    <div className={style.modalBackdrop} />
+    <CloseOnEscape handleClose={onClose}>
+      <FocusTrap>
+        <div aria-modal role="dialog" className={style.modal}>
```

Comment:
@megcrenshaw there's a semantic `<dialog>` html tag you could use for the dialogs, but it looks like you can't use `tabindex` which is used [here](https://github.com/code-dot-org/code-dot-org/blob/3c1f4d828edacbb697ce783be3e1f1ff49d66b49/apps/src/templates/MultipleSectionsAssigner.jsx#L164), and I'm not sure if that will require any more refactoring so this is not blocking. ([mdn web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog))

Reply:
Great callout! `dialog` should be used, but it's not compatible with FocusTrap (more info here https://github.com/code-dot-org/code-dot-org/pull/52290).

But I haven't tried yet using dialog *without* FocusTrap –– I'll investigate that. Thanks for raising!

Reply:
Cool, that def makes sense, and thanks for sharing the article!